cwl/src/main/scala/cwl/ScatterLogic.scala

```diff
@@ -195,30 +119,28 @@ object ScatterLogic {
 
     // We need to generate PBGONs for every output port of the call, so that they can be linked outside the scatter graph
     val portBasedGraphOutputNodes = callNode.outputPorts.map(op => PortBasedGraphOutputNode(op.identifier, op.womType, op))
-    
+
     def buildScatterNode(innerGraph: Graph, scatterProcessingFunction: ScatterProcessingFunction, scatterCollectingFunctionBuilder: ScatterCollectionFunctionBuilder) = {
       val scatterNodeBuilder = new ScatterNodeBuilder
       val outputPorts: Set[ScatterGathererPort] = innerGraph.nodes.collect { case gon: PortBasedGraphOutputNode =>
-        scatterNodeBuilder.makeOutputPort(scatterGatherPortTypeFunction(scatterMethod, scatterVariables)(gon.womType), gon)
+        scatterNodeBuilder.makeOutputPort(scatterGatherPortTypeFunction(scatterMethod, scatterVariablesNodes)(gon.womType), gon)
       }
 
-      val scatterNodeWithNewNodes = scatterNodeBuilder.build(innerGraph, outputPorts, scatterVariables.toList, scatterProcessingFunction, scatterCollectingFunctionBuilder)
-      knownNodes ++ scatterNodeWithNewNodes.nodes ++ stepInputFold.generatedNodes
+      scatterNodeBuilder.build(innerGraph, outputPorts, scatterVariablesNodes.toList, scatterProcessingFunction, scatterCollectingFunctionBuilder)
     }
-
-    // TODO POST 2.11 Could be forcomped in 2.12 but not in 2.11 because of the Checked
-    scatterProcessingFunctionCheck match {
-      case Right((scatterProcessingFunction, scatterCollectingFunctionBuilder)) =>
-        Graph.validateAndConstruct(Set(
-          callNodeAndNewNodes.node) ++ 
-          callNodeAndNewNodes.usedOuterGraphInputNodes ++ 
-          scatterVariables.toList ++ 
+    
+    for {
+      scatterProcessingFunctionAndBuilder <- scatterProcessingFunctionCheck
+      (scatterProcessingFunction, scatterCollectingFunctionBuilder) = scatterProcessingFunctionAndBuilder
```

Comment:
These two things are in a tuple, then they're unpacked from the tuple, then both of them are passed as consecutive arguments to `buildScatterNode` below. Maybe they should both just be wrapped in a case class or something?